### PR TITLE
[PLAT-8023] Prevent session/event delivery with invalid configuration

### PR DESCRIPTION
**Pull request by gingerbenw**
## Goal

If only one endpoint config option is set, no events or sessions should be sent from the app to avoid leaking data to SaaS Bugsnag.

## Changeset

Do not set delivery method when incorrect configuration is used

## Testing

Updated unit tests

### CODE REVIEW

File: packages/core/test/client.test.ts [hunk 1]
```diff
@@ -72,6 +72,46 @@ describe('@bugsnag/core/client', () => {
...
+
+      client.notify(new Error('111'), () => {}, (err, event) => {
+        expect(err).toBe(null)
+        expect(event).toBeTruthy()
+        expect(event.errors[0].errorMessage).toBe('111')
+        done()
+      })
+
+      // give the event loop a tick to see if the event gets sent
+      process.nextTick(() => done())
```
Comment (djskinner) [inline packages/core/test/client.test.ts:93]: not sure i understand this. won't this just cause the test to pass regardless of what assertions above fail?

Comment (gingerbenw) [reply packages/core/test/client.test.ts:93]: I'm not sure either - I copied it from a similar test but I think it needs to go altogether.

File: packages/core/test/client.test.ts [hunk 2]
```diff
@@ -72,6 +72,44 @@ describe('@bugsnag/core/client', () => {
...
+      client._setDelivery(client => ({
+        sendSession: () => { },
+        sendEvent: () => { done('sendEvent() should not be called') }
+      }))
+
+      client.notify(new Error('111'), undefined, (err, event) => {
+        expect(err).toBe(null)
+        expect(event).toBeTruthy()
+        expect(event.errors[0].errorMessage).toBe('111')
+        done()
```
Comment (djskinner) [inline packages/core/test/client.test.ts:110]: could we have a `expect(sendEvent).not.toHaveBeenCalled()` here?

Comment (gingerbenw) [reply packages/core/test/client.test.ts:110]: sure - I think it's redundant with `sendEvent: () => { done('sendEvent() should not be called') }` but it won't hurt.

**Review by gareththackeray — APPROVED** [review]

**Pull request merged**